src/Stratis.Bitcoin/P2P/Protocol/Payloads/GetProvenHeadersPayload.cs

```diff
@@ -0,0 +1,59 @@
+﻿using NBitcoin;
+
+namespace Stratis.Bitcoin.P2P.Protocol.Payloads
+{
+    /// <summary>
+    /// Get proven headers payload which requests proven headers using a similar mechanism as
+    /// the getheaders protocol message.
+    /// </summary>
+    /// <seealso cref="Stratis.Bitcoin.P2P.Protocol.Payloads.Payload" />
+    [Payload("getprovhdr")]
+    public class GetProvenHeadersPayload : Payload
+    {
+        /// <summary>
+        /// A block locator which represents a compact structure of one's chain position which can be used to find
+        /// forks with another chain.
+        /// </summary>
+        private BlockLocator blockLocator;
+
+        /// <summary>
+        /// Gets a block locator which represents a compact structure of one's chain position which can be used to find
+        /// forks with another chain.
+        /// </summary>
+        public BlockLocator BlockLocator
+        {
+            get => this.blockLocator;
+            set => this.blockLocator = value;
+        }
+
+        /// <summary>
+        /// Hash of the block after which constructing proven headers payload should stop.
+        /// </summary>
+        private uint256 hashStop = uint256.Zero;
+
+        /// <summary>
+        /// Gets a hash of the block after which constructing proven headers payload should stop.
```

Comment:
in order to avoid copypasting the comment probably place the comment here and at `private uint256 hashStop` use see ref tag